OpenRA.Mods.Common/Traits/GivesBounty.cs

```diff
@@ -38,26 +38,55 @@ class GivesBountyInfo : ITraitInfo
 		public object Create(ActorInitializer init) { return new GivesBounty(init.Self, this); }
 	}
 
-	class GivesBounty : INotifyKilled
+	class GivesBounty : INotifyKilled, INotifyCreated
 	{
 		readonly GivesBountyInfo info;
+		GainsExperience gainsExp;
+		Cargo cargo;
 
 		public GivesBounty(Actor self, GivesBountyInfo info)
 		{
 			this.info = info;
 		}
 
-		int GetMultiplier(Actor self)
+		void INotifyCreated.Created(Actor self)
+		{
+			gainsExp = self.TraitOrDefault<GainsExperience>();
+			cargo = self.TraitOrDefault<Cargo>();
+		}
+
+		// Returns 100's as 1, so as to keep accuracy for longer.
+		int GetMultiplier()
 		{
-			// returns 100's as 1, so as to keep accuracy for longer.
-			var gainsExp = self.TraitOrDefault<GainsExperience>();
 			if (gainsExp == null)
 				return 100;
 
 			var slevel = gainsExp.Level;
 			return (slevel > 0) ? slevel * info.LevelMod : 100;
 		}
 
+		int GetBountyValue(Actor self)
+		{
+			// Divide by 10000 because of GetMultiplier and info.Percentage.
+			return self.GetSellValue() * GetMultiplier() * info.Percentage / 10000;
+		}
+
+		int GetDisplayedBountyValue(Actor self)
+		{
+			var bounty = GetBountyValue(self);
+			if (cargo == null)
+				return bounty;
+
+			foreach (var a in cargo.Passengers)
+			{
+				var givesBounty = a.TraitOrDefault<GivesBounty>();
```

Comment:
This lookup is wrong. Since GivesBounty was updated with damagetypes, it's likely and valid that an actor will have more of this trait.